pulp-glue/pulp_glue/common/context.py

```diff
@@ -280,10 +280,12 @@ def call(
         return result
 
     @classmethod
-    def _check_task_finished(cls, task: EntityDefinition) -> bool:
+    def _check_task_finished(cls, task: EntityDefinition, expect_cancel: bool = False) -> bool:
         task_href = task["pulp_href"]
 
         if task["state"] == "completed":
+            if expect_cancel:
+                raise PulpException(_("The task {task_href} meant to be canceled, completed instead."))
```

Comment:
How about : The task completed before it got canceled?

Reply:
Discussed it with Lubos in the channel. This is what we agreed upon. Also i don't even see how this could happen on a recent pulpcore...